### ISSUE TITLE
[Bug]: PolygonSelector should hide itself when zoom/pan is active

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#28165]
### Bug summary

When interactive zoom/pan is active, widgets.PolygonSelector correctly disables itself (i.e. clicking and dragging zooms or pans the axes without creating a selection), but the selector cursor (a big circular marker) stays drawn at its last position (prior to zoom/pan being turned on).  It should instead be hidden.

### Code for reproduction

```Python
from pylab import *
ls = mpl.widgets.PolygonSelector(gca(), print)
# Move the cursor around, then toggle zoom pan via the toolbar
# or by pressing "o"/"p"
```

### Actual outcome

The PolygonSelector cursor stays drawn at an incorrect position.

### Expected outcome

The cursor is temporarily hidden.

### Additional information

Probably a good first issue?  I suspect the fix is along the lines of
```patch
diff --git i/lib/matplotlib/widgets.py w/lib/matplotlib/widgets.py
index eaa35e2544..4cd48ab3c1 100644
--- i/lib/matplotlib/widgets.py
+++ w/lib/matplotlib/widgets.py
@@ -4000,11 +4000,15 @@ class PolygonSelector(_SelectorWidget):
         # needs to process the move callback even if there is no button press.
         # _SelectorWidget.onmove include logic to ignore move event if
         # _eventpress is None.
-        if not self.ignore(event):
+        if self.ignore(event):
+            if not self.canvas.widgetlock.available(self) and self._xys:
+                self._xys[-1] = (np.nan, np.nan)
+                self._draw_polygon()
+            return False
+        else:
             event = self._clean_event(event)
             self._onmove(event)
             return True
-        return False
 
     def _onmove(self, event):
         """Cursor move event handler."""
```
i.e. if something else is active, just hide the last point of the polygon.  However this patch doesn't handle interaction with the various extra widget states (moving, etc.); likely it should only apply in the "position pending" mode (last else block of _onmove).

### Matplotlib Version

3.10.0.dev125+g63156f2ca6

### Matplotlib Backend

qtagg

### Python version

3.12

### Jupyter version

no

### Installation

git checkout

**Comment by Kaustbh** [comment]
Hi there! I'm new to open source, but I'm eager to dive in and help out. I came across this issue and I think I can tackle it, can I work on it ?

**Comment by ksunden** [comment]
Hi @Kaustbh,

We do not assign any issues, and anybody is free to work on anything. In general, we tend to give preference to the first person to open a PR (absent a clear technical reason to opt for a newer PR).

In this case, there are no PRs yet for this issue.

**Comment by Kaustbh** [comment]
@anntzer , I tried to reproduce the issue , can you tell what is the problem . (backend = WebAgg) ,

https://github.com/matplotlib/matplotlib/assets/97254178/71fffd61-e8bf-42d2-a608-aba74122c942

**Comment by Kaustbh** [comment]
@anntzer  please reply .

**Comment by anntzer** [comment]
Your movie doesn't play, and as noted in the report this is with qtagg (or likely tkagg should work too).

**Comment by QuLogic** [comment]
Fixed by #28262.
